src/DssSpell.t.sol

```diff
@@ -859,4 +890,66 @@ contract DssSpellTest is DssSpellTestBase {
     }
 
     // SPELL-SPECIFIC TESTS GO BELOW
+
+    function testDirectSparkMorphoIntegration() public {
+        bytes32 ilk         = "DIRECT-SPARK-MORPHO-DAI";
+        address vault       = 0x73e65DBD630f90604062f6E02fAb9138e713edD9;
+        address operator    = 0x298b375f24CeDb45e936D7e21d6Eb05e344adFb5;
+        uint256 debtCeiling = 100 * MILLION * WAD;
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        D3MHubLike hub = D3MHubLike(addr.addr("DIRECT_HUB"));
+        D3MMomLike mom = D3MMomLike(addr.addr("DIRECT_MOM"));
+        D3M4626PoolLike pool = D3M4626PoolLike(addr.addr("DIRECT_SPARK_MORPHO_DAI_POOL"));
+        D3MOperatorPlanLike plan = D3MOperatorPlanLike(addr.addr("DIRECT_SPARK_MORPHO_DAI_PLAN"));
+        D3MOracleLike oracle = D3MOracleLike(addr.addr("DIRECT_SPARK_MORPHO_DAI_ORACLE"));
+
+        { // Do a bunch of sanity checks of the values that were set in the spell
+            (address _pool, address _plan, uint256 tau, uint256 _culled,) = hub.ilks(ilk);
+            assertEq(_pool, address(pool));
+            assertEq(_plan, address(plan));
+            assertEq(tau, 7 days);
+            assertEq(_culled, 0);
+            assertEq(hub.vow(), address(vow));
+            assertEq(hub.end(), address(end));
+            assertEq(mom.authority(), address(chief));
+            assertEq(pool.vault(), vault);
+            assertEq(pool.redeemable(), vault);
+            assertEq(plan.operator(), operator);
+            assertEq(plan.wards(address(mom)), 1);
+            assertEq(plan.active(), true);
+            assertEq(oracle.hub(), address(hub));
+            (address pip,) = spotter.ilks(ilk);
+            assertEq(pip, address(oracle));
+            assertEq(vat.wards(address(hub)), 1);
+        }
+
+        // Set supply queue
+        bytes32[] memory newSupplyQueue = new bytes32[](1);
+        newSupplyQueue[0] = 0x57f4e42c0707d3ae0ae39c9343dcba78ff79fa663da040eca45717a9b0b0557f; // Idle DAI market
+        vm.prank(operator); IMetaMorpho(vault).setSupplyQueue(newSupplyQueue);
```

Comment:
Please add an explanation of where the bytes32 value comes from and why it is needed

Reply:
Basically, setting the supply queue is required for the proper functioning of the vault just like calling `setTargetAssets` on a plan. The specific `0x57f4...` value is provided by the spark [here](https://github.com/makerdao/spells-mainnet/pull/402#discussion_r1536332171).

On a technical side, I believe this market id is generated via [keccak256 hash of the MarketParams object](https://github.com/morpho-org/morpho-blue/blob/e262e84606e34943955c425af1bd33c86bad5cf1/src/libraries/MarketParamsLib.sol#L18) and is created in advance. In our case it's created via [this transaction](https://etherscan.io/tx/0x37d381d56f280ce046e285b54db947068bd0220ff483175fb921b3ecec0a09ea) which can be found on a [page which lists all previously created Morpho markets](https://dune.com/queries/3330733/5580054).

Decoding the calldata would provide values used to generate this id, which later can be recreated this way:
```solidity
// Values decoded from https://etherscan.io/tx/0x37d381d56f280ce046e285b54db947068bd0220ff483175fb921b3ecec0a09ea
MarketParams memory marketParams = MarketParams({
    loanToken: 0x6B175474E89094C44Da98b954EedeAC495271d0F,
    collateralToken: address(0),
    oracle: address(0),
    irm: address(0),
    lltv: 0
});

// Snippet taken from https://github.com/morpho-org/morpho-blue/blob/e262e84606e34943955c425af1bd33c86bad5cf1/src/libraries/MarketParamsLib.sol#L18
bytes32 marketId;
uint256 MARKET_PARAMS_BYTES_LENGTH = 5 * 32;
assembly ("memory-safe") {
    marketId := keccak256(marketParams, MARKET_PARAMS_BYTES_LENGTH)
}

// Value taken from https://github.com/makerdao/spells-mainnet/pull/402#discussion_r1536332171
assertEq(marketId, 0x57f4e42c0707d3ae0ae39c9343dcba78ff79fa663da040eca45717a9b0b0557f, 'TestError/id does not match');
```

TLDR: I've added a comment explaining where is this value from.